mypy/fastparse.py

```diff
@@ -279,6 +279,13 @@ def parse(
         # Disable deprecation warnings about \u
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore", category=DeprecationWarning)
+            if options.allow_commented_type_annotations:
+                lines = str(source).split("\n")
+                result = ""
+                for line in lines:
+                    if not str(line).startswith("# type:"):
```

Comment:
This is sorting out all lines that don't start with `# type:`? Typically `# type: ...` appears at the end of a line, not on its own at the start.